pkg/hosts/hosts.go

```diff
@@ -2,27 +2,49 @@ package hosts
 
 import (
 	"fmt"
+	"regexp"
 	"sort"
 	"strings"
 
 	"github.com/goodhosts/hostsfile"
 )
 
+const (
+	ClusterDomain = ".crc.testing"
+	AppsDomain    = ".apps-crc.testing"
+)
+
 type Hosts struct {
-	File *hostsfile.Hosts
+	File       *hostsfile.Hosts
+	HostFilter func(string) bool
 }
 
 func New() (*Hosts, error) {
 	file, err := hostsfile.NewHosts()
 	if err != nil {
 		return nil, err
 	}
+
 	return &Hosts{
-		File: &file,
+		File:       &file,
+		HostFilter: defaultFilter,
 	}, nil
 }
 
+func defaultFilter(s string) bool {
+	return suffixFilter(ClusterDomain, s) || suffixFilter(AppsDomain, s)
+}
+
+func suffixFilter(suffix, s string) bool {
+	regexp := regexp.MustCompile("^[a-z0-9]([-a-z0-9]*[a-z0-9])?" + regexp.QuoteMeta(suffix) + "$")
```

Comment:
This won't allow `foo.bar.apps-crc.testing`, but I don't know if such names can be created in a cluster? Are we sure the names will always be lower case? Or could we get "FOO.apps-crc.testing"?

Reply:
I'd also try to cache the `regexp.MustCompile` result, feels wasteful to always recreate it.

Reply:
Capital letter are forbidden: 

```
# * spec.host: Invalid value: "Downloads-openshift-console.apps-crc.testing": host must conform to DNS 1123 naming conventions: [spec.host: Invalid value: "Downloads-openshift-console.apps-crc.testing": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')]
``` 

Sub DNS names are allowed. I am fixing this.

Reply:
all done!